core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -161,8 +161,8 @@ class Partition(val topic: String,
    *  and setting the new leader and ISR
    */
   def makeLeader(controllerId: Int,
-                 partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
-    inWriteLock(leaderIsrUpdateLock) {
+                 partitionStateInfo: PartitionStateInfo, correlationId: Int) = {
```

Comment:
Do you mean to return `Unit`? If so, you should remove the `=`. Inferring the result type is something to be avoided for public methods.